tests/asahirov/test_hw7_retry.py

```diff
@@ -0,0 +1,42 @@
+"""
+Create decorator, which will retry inner function multiple times untill it passes
+"""
+import random
+import pytest
+
+
+def retry_decorator(retries: int = 5, exception_types: tuple = (Exception,)):
+    def retry(func):
+        def wrapper(*args, **kwargs):
+            error = None
+            for i in range(retries):
+                try:
+                    return func(*args, **kwargs)
+                except exception_types as e:
+                    error = e
+                    print(f"Function failed. Retries count: #{i + 1}: {e}")
+            if error:
+                raise error
+        return wrapper
+    return retry
```

Comment:
raise in same exception block where caught
not sure if it makes difference but it is more clean

```

def retry_decorator(retries: int = 5, exception_types: tuple = (Exception,)):
    def retry(func):
        
        def wrapper(*args, **kwargs):

            for i in range(retries):
                try:
                    return func(*args, **kwargs)
                except exception_types as e:
                    print(f"Function failed. Retries count: #{i + 1}: {e}")
                    if i == retries - 1:
                        raise      # reraises active error

        return wrapper
    return retry
```